api/envoy/config/filter/http/transformation/v2/transformation_filter.proto

```diff
@@ -153,24 +153,55 @@ message Transformation {
 // Extractions can be used to extract information from the request/response.
 // The extracted information can then be referenced in template fields.
 message Extraction {
+  // The mode of operation for the extraction.
+  enum Mode {
+    // Default mode. Extract the value of the subgroup-th capturing group.
+    EXTRACT = 0; 
+    // Replace the value of the subgroup-th capturing group with the replacement_text.
+    // Note: replacement_text must be set for this mode.
+    SINGLE_REPLACE = 1;
+    // Replace all matches of the regex in the source with the replacement_text.
+    // Note: replacement_text must be set for this mode.
+    // Note: subgroup is ignored for this mode. configuration will fail if subgroup is set.
+    // Note: restrictions on the regex are different for this mode. See the regex field for more details.
+    REPLACE_ALL = 2;
+  }
 
   // The source of the extraction
   oneof source {
     // Extract information from headers
     string header = 1;
     // Extract information from the request/response body
     google.protobuf.Empty body = 4;
+    // Extract information from dynamic metadata
```

Comment:
I need to update this comment -- this value is used as a key in dynamic metadata and this comment needs to express that clearly

Reply:
No longer relevant, as the functionality adding support for dynamic metadata extractor inputs has been removed from this PR